File: src/main/java/ukalus/container/PropertySetter.java
package ukalus.container;

import java.lang.reflect.Method;
import java.util.HashMap;
import java.util.Map;
import java.util.stream.Stream;

public class PropertySetter {

  private Map<String, Resolver> properties;

  public PropertySetter(Map<String, Resolver> properties) {
    this.properties = properties;
  }

  public void setProperties(Object input) throws IllegalArgumentException {
    Method[] methods = input.getClass().getMethods();
    Map<String, Method> setters = new HashMap<>(methods.length);
    Stream.of(methods).forEach(new SetterCollector(setters));
    properties.entrySet().forEach(new SetProperty(setters, input));
  }
}

File: src/main/java/ukalus/container/PropertySetter.kt
package ukalus.container

import java.lang.reflect.Method
import java.util.HashMap
import java.util.stream.Stream

class PropertySetter(private val properties: Map<String, Resolver>) {

    fun setProperties(input: Any) {
        val methods = input.javaClass.methods
        val setters = HashMap<String, Method>(methods.size)
        Stream.of(*methods).forEach(SetterCollector(setters))
        properties.entries.forEach(setProperty(setters, input))
    }
}
